gateway/src/integration-test/java/com/hurence/webapiservice/http/grafana/ImportJsonEndPointIT.java

```diff
@@ -0,0 +1,109 @@
+                "/http/grafana/query/extract-algo/testAdd/request.json",
+                "/http/grafana/query/extract-algo/testAdd/expectedResponse.json");
+        assertRequestGiveResponseFromFile(vertx, testContext,
+                "/http/grafana/query/extract-algo/test0/request.json",
+                "/http/grafana/query/extract-algo/test0/expectedResponse.json");
+    }
+
+    @Test
+    @Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+    public void testWrongAdd(Vertx vertx, VertxTestContext testContext) {
+
+        assertWrongAddRequestGiveResponseFromFile(vertx, testContext,
+                "/http/grafana/query/extract-algo/testWrongAdd/request.json");
+    }
+
+    public void assertRequestGiveResponseFromFile(Vertx vertx, VertxTestContext testContext,
+                                                  String requestFile, String responseFile) {
+        assertHelper.assertRequestGiveArrayResponseFromFile(vertx, testContext, requestFile, responseFile);
+    }
+
+    public void assertAddRequestGiveResponseFromFile(Vertx vertx, VertxTestContext testContext,
+                                                  String requestFile, String responseFile) {
+        final FileSystem fs = vertx.fileSystem();
+        Buffer requestBuffer = fs.readFileBlocking(AssertResponseGivenRequestHelper.class.getResource(requestFile).getFile());
+        webClient.post("/historian-server/ingestion/json")
+                .as(BodyCodec.jsonObject())
+                .sendBuffer(requestBuffer.getDelegate(), testContext.succeeding(rsp -> {
+                    testContext.verify(() -> {
+                        assertEquals(200, rsp.statusCode());
+                        assertEquals("OK", rsp.statusMessage());
+                        JsonObject body = rsp.body();
+                        Buffer fileContent = fs.readFileBlocking(AssertResponseGivenRequestHelper.class.getResource(responseFile).getFile());
+                        JsonObject expectedBody = new JsonObject(fileContent.getDelegate());
+                        assertEquals(expectedBody, body);
+                        testContext.completeNow();
+                    });
+                }));
+    }
+    public void assertWrongAddRequestGiveResponseFromFile(Vertx vertx, VertxTestContext testContext,
+                                                     String requestFile) {
+        final FileSystem fs = vertx.fileSystem();
+        Buffer requestBuffer = fs.readFileBlocking(AssertResponseGivenRequestHelper.class.getResource(requestFile).getFile());
+        webClient.post("/historian-server/ingestion/json")
+                .sendBuffer(requestBuffer.getDelegate(), testContext.succeeding(rsp -> {
+                    testContext.verify(() -> {
+                        assertEquals(500, rsp.statusCode());
+                        assertEquals("Internal Server Error", rsp.statusMessage());
+                        testContext.completeNow();
+                    });
+                }));
+    }
+}
```

Comment:
Same here, I do not think a method that do that exist, but you could move this function which is general in another place where it can be used again. Maybe a method where the status code, the message and the endpoint are parameters ?

Reply:
Well you could keep this method but then the method would use the more general method.

Reply:
done, i added a new method in the class that containes the general method, one that checks the worng request will get 500 code status